### PR TITLE
[agent]: Turn on components for Backhaul STA Steering

### LINKED ISSUE #408: [TASK] Backhaul STA steering
## Description

The agent should perform steering of the backhaul STA interface based on the controller's backhaul STA steering request.

## Exit criteria

- BWL implementation of backhaul STA steering on nl80211.
- BWL implementation of backhaul STA steering on dummy.
- Agent does backhaul steering according to policy message.
- Test flow for the above.
- Manually tested on Turris Omnia.

### PR DESCRIPTION
**Pull request by VladyslavTupikin**
According to the UML diagram from PR #1419 need to add Bachaul STA Steering TLV's to the builds, update method parseNextTlv in CmduMessageRx - add cases with new TLV's, add stubs in the HAL for associate method.

### CODE REVIEW

**Comment by VladyslavTupikin** [comment]
PR should be merged after https://github.com/prplfoundation/prplMesh/pull/1419

**Comment by VladyslavTupikin** [comment]
PR #1419 merged. This PR can be reviewed.

File: common/beerocks/bwl/include/bwl/sta_wlan_hal.h [hunk 1]
```diff
@@ -48,6 +48,8 @@ class sta_wlan_hal : public virtual base_wlan_hal {
                          bool mem_only_psk, const std::string &bssid, uint8_t channel,
                          bool hidden_ssid) = 0;
 
+    virtual bool associate(const std::string &bssid, uint8_t channel) = 0;
```
Comment (arnout) [inline common/beerocks/bwl/include/bwl/sta_wlan_hal.h]: Please don't use strings for mac addresses in new functions.
```suggestion
    virtual bool associate(const sMacAddr &bssid, uint8_t channel) = 0;
```
Yes, it requires an additional string-to-mac conversion in the caller, and then a mac-to-string conversion to send it to wpa_supplicant, but this is the only way we can ever hope to stamp out the evil use of strings for mac addresses.

File: common/beerocks/bwl/dummy/sta_wlan_hal_dummy.cpp [hunk 2]
```diff
@@ -41,6 +41,8 @@ bool sta_wlan_hal_dummy::connect(const std::string &ssid, const std::string &pas
     return true;
 }
 
+bool sta_wlan_hal_dummy::associate(const std::string &bssid, uint8_t channel) { return true; }
```
Comment (arnout) [inline common/beerocks/bwl/dummy/sta_wlan_hal_dummy.cpp]: I think it's useful to already add a log message in each of the implementations, even if it's currently unimplemented.

**Review by arnout — APPROVED** [review]
I actually meant to also add the logic that handles the steering request and makes the bwl call, but in fact it makes sense to merge the current state.

File: common/beerocks/bwl/include/bwl/sta_wlan_hal.h (hunk 1 above)
Comment (VladyslavTupikin) [reply common/beerocks/bwl/include/bwl/sta_wlan_hal.h]: fixed

File: common/beerocks/bwl/dummy/sta_wlan_hal_dummy.cpp (hunk 2 above)
Comment (VladyslavTupikin) [reply common/beerocks/bwl/dummy/sta_wlan_hal_dummy.cpp]: added debug prints

**Review by abelog — APPROVED** [review]
LGTM

File: common/beerocks/bwl/include/bwl/sta_wlan_hal.h [hunk 3]
```diff
@@ -48,6 +48,8 @@ class sta_wlan_hal : public virtual base_wlan_hal {
                          bool mem_only_psk, const std::string &bssid, uint8_t channel,
                          bool hidden_ssid) = 0;
 
+    virtual bool associate(const sMacAddr &bssid, uint8_t channel) = 0;
```
Comment (arnout) [inline common/beerocks/bwl/include/bwl/sta_wlan_hal.h]: I just noticed now: the roam() function, three lines below, is in fact exactly what we need, so this commit is not needed at all.

I've removed the Ready for Merge label so you can remove this commit.

Comment (VladyslavTupikin) [reply common/beerocks/bwl/include/bwl/sta_wlan_hal.h]: removed useless commit

**Pull request merged**